lib/assertions.js

```diff
@@ -766,39 +766,61 @@ module.exports = (expect) => {
     );
   });
 
-  expect.addAssertion('<array-like> [not] to contain <any+>', function (
-    expect,
-    subject
-  ) {
-    const args = Array.prototype.slice.call(arguments, 2);
-    expect.withError(
-      () => {
-        args.forEach((arg) => {
-          expect(
-            subject &&
-              Array.prototype.some.call(subject, (item) =>
-                expect.equal(item, arg)
-              ),
-            '[not] to be truthy'
-          );
-        });
-      },
-      (e) => {
-        expect.fail({
-          diff:
-            expect.flags.not &&
-            ((output, diff, inspect, equal) =>
-              diff(
-                subject,
-                Array.prototype.filter.call(
-                  subject,
-                  (item) => !args.some((arg) => equal(item, arg))
-                )
-              )),
-        });
+  expect.addAssertion(
+    '<array-like> [not] to [only] contain <any+>',
+    function (expect, subject) {
+      const args = Array.prototype.slice.call(arguments, 2);
+
+      if (expect.flags.only && expect.flags.not) {
+        expect.errorMode = 'bubble';
+        expect.fail(
+          'The "not" flag cannot be used together with "to only contain".'
+        );
       }
-    );
-  });
+
+      expect.withError(
+        () => {
+          args.forEach((arg) => {
+            expect(
+              subject &&
+                Array.prototype.some.call(subject, (item) =>
+                  expect.equal(item, arg)
+                ),
+              '[not] to be truthy'
+            );
+          });
+          if (expect.flags.only) {
+            expect(subject.length, 'to be', args.length);
```

Comment:
This check is "defeated" when the subject contains duplicate items:

```
it('should handle duplicate items', () => {
  expect(
    [{ foo: 123 }, { foo: 123 }, { bar: 456 }],
    'to only contain',
    { foo: 123 },
    { bar: 456 }
  );
});
```

Results in a diff-less diff:

```js
expected [ { foo: 123 }, { foo: 123 }, { bar: 456 } ]
to only contain { foo: 123 }, { bar: 456 }

[
  { foo: 123 },
  { foo: 123 },
  { bar: 456 }
]
```

Reply:
That's an interesting case! It might be that what I'm looking for is not `to only contain` anyway. Your example is obviously true, but what I actually need, is more like: 

```js
expect(myArray, 'to be made out of items in the following array in any order', [1,2,3,4,5,6]);
```

Going on from your example `expect([1,1,1,1,1], `to only contain`, 1);` should obviously also pass. It's a much different implementation we need than what I have proposed here in this PR. We need to count unique values (not really - but in the structural comparison way we do where `{ foo: 123}` is equal to `{ foo:123 }`, and make sure that it's only values from that set which make up the LHS array.

It feels like a disqualifier for this entire approach (and the original use-case that I had).

Reply:
You need to run through the RHS items, for each of them you check if it matches each of the items in the LHS. You record that information in an array of boolean, where you always or the results. If the RHS item doesn't match any items you fail. When you are done you check that every item in the array is true; otherwise you fail.

I think the above would work, but I also realise that I can't ask you to go an do that, so I'll probably volunteer to do it. But I guess you need to solution now, so that is a bit of pickle 😅

Reply:
I can also see that I lied to you earlier about `to contain` using `to satisfy` sematics, it looks like it is `to equal` semantics - which I'm by the way pretty happy about because it is less surprising.

Reply:
@gustavnikolaj, regardless of your original use case, I still think this would be a nice feature, so I hope we can get it in 🤗

Reply:
Just for the record, there's no stress, I have solved my problem and I'm not blocked on this, so we can take our time. 

I ended up just sorting my array to force it to be deterministic. :-)

Reply:
I've taken a stab at an implementation adressing this - and there's a new review thread there, so I'll be resolving this thread.